contracts/core/Pool.sol

```diff
@@ -282,6 +314,7 @@ contract Pool is IPool, PoolERC20 {
 
   //Assumption: token0 is ETH, so when you transfer to the user, always transfer token1
   function swapExactInEthForToken(address to) external payable onlyEthPool nonZeroAddress(to) {
+    // don't assume but should add a requirement checking token0 is ETH in the constructor
```

Comment:
The factory always produce an ETH-Token pool with ETH as token0

Reply:
Ah okay. Then the comment is not exactly gud. Should say like it's guaranteed token0 is ETH (by the factory)